### ISSUE TITLE
Uncaught TypeError: Cannot read property 'localeCompare' of undefined

### ISSUE THREAD

**Issue by subhajitwiner** [atom/atom#19017]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.35.1 x64
**Electron**: 2.0.18
**OS**: Microsoft Windows 10 Pro
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'localeCompare' of undefined

```
At C:\Users\subha\AppData\Local\atom\app-1.35.1\resources\app\static\<embedded>:11

TypeError: Cannot read property 'localeCompare' of undefined
    at i (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:641982)
    at Array.sort (native)
    at SnippetsProvider.findSuggestionsForPrefix (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:641843)
    at SnippetsProvider.getSuggestions (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:641437)
    at n.forEach.t (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:631266)
    at Array.forEach (<anonymous>)
    at AutocompleteManager.getSuggestionsFromProviders (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:630782)
    at AutocompleteManager.findSuggestions (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:630520)
    at AutocompleteManager.requestNewSuggestions (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:639053)
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:640144)
    at Function.simpleDispatch (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:1172354)
    at Emitter.emit (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:1173795)
    at TextBuffer.emitDidChangeTextEvent (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:504122)
    at TextBuffer.transact (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:492297)
    at TextEditor.transact (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:24215)
    at mergeIntersectingSelections (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:16150)
    at TextEditor.mergeSelections (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:39484)
    at TextEditor.mergeIntersectingSelections (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:39038)
    at TextEditor.mutateSelectedText (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:16113)
    at TextEditor.insertText (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:15662)
    at TextEditor.e.(anonymous function) [as insertText] (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:259365)
    at TextEditorComponent.didTextInput (~/AppData/Local/atom/app-1.35.1/resources/app/static/<embedded>:11:93878)
```

### Commands

```
     -1:01.0 editor:newline (input.hidden-input)
     -0:59.2.0 core:paste (input.hidden-input)
  2x -0:58.4.0 core:move-left (input.hidden-input)
     -0:57.6.0 core:move-right (input.hidden-input)
     -0:53.9.0 core:move-left (input.hidden-input)
     -0:38.4.0 core:paste (input.hidden-input)
     -0:37 core:move-right (input.hidden-input)
     -0:28.4.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
angular-bootstrap 1.0.1 
atom-bootstrap4 1.4.0 
autocomplete-jquery-mobile 0.2.2 
bootstrap-sass-snippets 2.1.1 
codeigniter 0.4.0 
codeigniter-nucleus 0.4.3 
drunken-php 0.2.4 
html-to-javascript 0.2.0 
ide-php 0.7.18 
jQuery 1.2.6 
Quick-JavaScript 2.2.0 
turbo-javascript 1.4.0 
v-bootstrap4 0.2.2 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* What were you doing at the time of the error?
* Is the error is reproducible?
* If reproducible, do you see the error in safe mode (`atom --safe`)?

**Comment by eelstork** [comment]
I encountered this error and clearly this results from an error in the snippets file (user error?) or how its content is handled.
I had a snippets file that validated correctly and mostly worked but typing the letter 'u' would cause the error. I defined 3 custom snippets starting with 'u'
Deleting the 3 snippets and re-typing them manually fixed the error.
Unfortunately didn't save the offending snippets file. Will update if the error recurs.
